auth-tokens-filter/src/main/java/com/palantir/tokens/auth/http/BasicAuthToBearerTokenFilter.java

```diff
@@ -0,0 +1,135 @@
+    private static final String BASIC_AUTH_STR = "Basic";
+    private static final Logger log = LoggerFactory.getLogger(BasicAuthToBearerTokenFilter.class);
+    private static final BaseEncoding BASE_64_ENCODING = BaseEncoding.base64Url();
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+    }
+
+    @Override
+    public final void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+        ServletRequest updatedRequest;
+        try {
+            updatedRequest = getRequestWithToken(request);
+        } catch (NullRawAuthHeaderException e) {
+            updatedRequest = request;
+        } catch (AuthHeaderNotBasicAuthException e) {
+            log.warn("AuthHeaderNotBasicAuthException", e);
+            updatedRequest = request;
+        }
+        chain.doFilter(updatedRequest, response);
+    }
+
+    @Override
+    public void destroy() {
+    }
+
+    private ServletRequest getRequestWithToken(ServletRequest request)
+            throws AuthHeaderNotBasicAuthException, NullRawAuthHeaderException {
+        if (request instanceof HttpServletRequest) {
+            final HttpServletRequest httpRequest = (HttpServletRequest) request;
+            final String rawAuthHeader = httpRequest.getHeader(HttpHeaders.AUTHORIZATION);
+            return getRequestWithTokenFromRawAuthHeader(rawAuthHeader, httpRequest);
+        } else {
+            throw new AuthHeaderNotBasicAuthException("Request is not an HttpServletRequest.");
+        }
+    }
+
+    private ServletRequest getRequestWithTokenFromRawAuthHeader(String rawAuthHeader, HttpServletRequest request)
+            throws AuthHeaderNotBasicAuthException, NullRawAuthHeaderException {
+        if (rawAuthHeader == null) {
+            throw new NullRawAuthHeaderException("Raw auth header is null.");
+        } else if (isBasicAuth(rawAuthHeader)) {
+            final AuthHeader authHeader = getAuthHeader(rawAuthHeader);
+            return getRequestWithTokenFromAuthHeader(authHeader, request);
+        } else {
+            throw new AuthHeaderNotBasicAuthException("Auth header is not basic auth.");
+        }
+    }
+
+    private ServletRequest getRequestWithTokenFromAuthHeader(final AuthHeader authHeader, HttpServletRequest request) {
+        return new HttpServletRequestWrapper(request) {
+            @Override
+            public String getHeader(String name) {
+                if (Objects.equals(name, HttpHeaders.AUTHORIZATION)) {
+                    return authHeader.toString();
+                } else {
+                    return super.getHeader(name);
+                }
+            }
+        };
+    }
+
+    private AuthHeader getAuthHeader(String rawAuthHeader) {
+        final String password = getPassword(rawAuthHeader);
+        return AuthHeader.valueOf(password);
+    }
+
+    private String getPassword(String rawAuthHeader) {
```

Comment:
I'm happy to inline `isBasicAuth`, but I think `getPassword` is large enough to warrant its own method.